core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewSystemTable.java

```diff
@@ -0,0 +1,132 @@
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.trino.connector.system.jdbc.FilterUtil.tablePrefix;
+import static io.trino.connector.system.jdbc.FilterUtil.tryGetSingleVarcharValue;
+import static io.trino.metadata.MetadataListing.getMaterializedViews;
+import static io.trino.metadata.MetadataListing.listCatalogs;
+import static io.trino.metadata.MetadataListing.listSchemas;
+import static io.trino.spi.connector.SystemTable.Distribution.SINGLE_COORDINATOR;
+import static java.util.Objects.requireNonNull;
+
+public class MaterializedViewSystemTable
+        implements SystemTable
+{
+    private final Metadata metadata;
+    private final AccessControl accessControl;
+    private final MaterializedViewRowAdapter materializedViewAdapter;
+
+    @Inject
+    public MaterializedViewSystemTable(
+            Metadata metadata,
+            AccessControl accessControl,
+            MaterializedViewRowAdapter materializedViewAdapter)
+    {
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.accessControl = requireNonNull(accessControl, "accessControl is null");
+        this.materializedViewAdapter = requireNonNull(materializedViewAdapter, "materializedViewAdapter is null");
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return materializedViewAdapter.getTableLayout();
+    }
+
+    @Override
+    public RecordCursor cursor(
+            ConnectorTransactionHandle transactionHandle,
+            ConnectorSession connectorSession,
+            TupleDomain<Integer> constraint)
+    {
+        Session session = ((FullConnectorSession) connectorSession).getSession();
+
+        List<MaterializedViewHandle> materializedViews = getAllMaterializedViews(session, constraint);
+        List<Object[]> materializedViewRows = materializedViewAdapter.toTableRows(materializedViews);
+
+        InMemoryRecordSet.Builder displayTable = InMemoryRecordSet.builder(getTableMetadata());
+        materializedViewRows.forEach(displayTable::addRow);
+
+        return displayTable.build().cursor();
+    }
+
+    private List<MaterializedViewHandle> getAllMaterializedViews(Session session, TupleDomain<Integer> constraint)
+    {
+        ImmutableList.Builder<MaterializedViewHandle> materializedViews = ImmutableList.builder();
+
+        Optional<String> catalogNameFilter = tryGetSingleVarcharValue(constraint, 0);
+        Optional<String> schemaNameFilter = tryGetSingleVarcharValue(constraint, 1);
+        Optional<String> tableNameFilter = tryGetSingleVarcharValue(constraint, 2);
+
+        for (String catalogName : listCatalogs(session, metadata, accessControl, catalogNameFilter).keySet()) {
+            for (String schemaName : listSchemas(session, metadata, accessControl, catalogName, schemaNameFilter)) {
```

Comment:
We actually don't need `listSchemas` loop.
Just use `QualifiedTablePrefix materializedViewPrefix = tablePrefix(catalogName, schemaFilter, tableFilter)`